### ISSUE TITLE
Framework laptop, set VSCode to default Wayland use such that fractional scaling looks good

### ISSUE THREAD

**Issue by frederikb96** [ublue-os/bluefin#1655]
### Describe the bug

I have a fresh bluefin on my framework 13 laptop. I start vscode and it looks blurry. I checked web for the issue and found this closed one here: #1362
I followed the workaround to set `ELECTRON_OZONE_PLATFORM_HINT=auto` and then it works.

My request would be to set this out of the box for Framework laptops such that users dont have to check the web to find this issue and apply the workaround themselfes. Thanks :)

### What did you expect to happen?

On fresh framework laptop, a non blurry VS Code :)

**Comment by zackify** [comment]
yeah vs code is unusable for me on fw13 with 150% scale, 125% its still bad

**Comment by castrojo** [comment]
Yeah we _could_ override the code.desktop but we're hesitant to do this per app since there's almost always some unintended consequence or config that users might be doing.

For now we recommend copying `/usr/share/applications/code.desktop` into `~.local/share/applications` and then making the change in that file and then that'll sort it.

**Comment by zackify** [comment]
we should add this to the vs code docs or framework section

**Comment by castrojo** [comment]
This should also be solved by F41, which is a few weeks out, and then maybe we recommend :stable for framework from then on.

**Comment by frederikb96** [comment]
Stumbled across this again, since VS Code is now working for the first time very nicely with Wayland support enabled, thanks to this issue being solved, where pane moving was not possible yet due to upstream Electron problem: https://github.com/microsoft/vscode/issues/156723

This also fixed pane movement for Obsidian and Wayland. So from now on, I started to use fractional scaling, and it is wonderful :D

I also wrote a small Ansible script to apply it to VSCode directly so dont have to do it manually for my maschines, especially when desktop launcher updates due to VSCode update :) https://github.com/frederikb96/vscode-wayland-support-enable-desktop-file

**Comment by nickdiego** [comment]
FWIW, at Chromium side, we've worked recently on proper fractional scaling support (addressing the blurriness issues reported) which started landing since Chromium M128, behind the `WaylandPerSurfaceScale` feature flag. The work was tracked at [1] (further technical details at [2]). I'm not sure when/if it's properly plumbed in electron/vscode though.

- [1] https://issues.chromium.org/issues/336007385
- [2] https://notes.nickdiego.dev/chromium/proposal/wayland-per-window-scaling

**Comment by frederikb96** [comment]
I use VS Code every day with the flags `--enable-features=UseOzonePlatform --ozone-platform=wayland'` and fractional scaling turned on. Haven't had any issues since then.
